src/libraries/Common/src/Interop/Unix/System.Native/Interop.CopyFile.cs

```diff
@@ -9,7 +9,28 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
+        private static volatile int s_tryCopyFileRange;
+
+        private static bool TryCopyFileRange
```

Comment:
I think it would be better to have this whole thing in the shim. 

Try to call `copy_file_range`. If it returns `ENOSYS`, remember it in a static bool and do not call it next time.

Reply:
> Try to call copy_file_range. If it returns ENOSYS, remember it in a static bool and do not call it next time.

Ah ok, this won't work for "Try to call copy_file_range. If it returns ENOSYS, remember it in a static bool and do not call it next time."

Still, I would keep the entire logic in the native shim.

Reply:
That was my first thought as well. Then I checked how is the version parsed (https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Environment.OSVersion.Unix.cs) and came to the conclusion that duplicating the parsing code may not be a good tradeoff.

Reply:
Yes, I put it here so I could reuse the managed parsing code. I can add it in native code.

Reply:
You do not need to do a full parsing to do a simple version check. Something like:
```
bool IsAtLeastRelease53()
{
    struct utsname name;
    if (uname(&name) != 0)
        return false;
    return (name.release.version[0] > '5' || 
        (name.release.version[0] == '5` && (name.release.version[1] != '.' || name.release.version[2] >= '3')));
}
```